clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1966,22 +2003,39 @@ class SyclKernelPointerHandler : public SyclKernelFieldHandler {
     return true;
   }
 
-  bool handlePointerType(FieldDecl *FD, QualType FieldTy) final {
-    QualType PointeeTy = FieldTy->getPointeeType();
-    Qualifiers Quals = PointeeTy.getQualifiers();
-    LangAS AS = Quals.getAddressSpace();
-    // Leave global_device and global_host address spaces as is to help FPGA
-    // device in memory allocations.
-    if (!PointeeTy->isFunctionType() && AS != LangAS::sycl_global_device &&
-        AS != LangAS::sycl_global_host)
-      Quals.setAddressSpace(LangAS::sycl_global);
-    PointeeTy = SemaRef.getASTContext().getQualifiedType(
-        PointeeTy.getUnqualifiedType(), Quals);
-    QualType ModTy = SemaRef.getASTContext().getPointerType(PointeeTy);
-    addField(FD, ModTy);
+  bool leaveArray(FieldDecl *FD, QualType ArrayTy, QualType ET) final {
+    QualType ModifiedArrayElement = ModifiedArrayElementsOrArray.pop_back_val();
+
+    const ConstantArrayType *CAT =
+        SemaRef.getASTContext().getAsConstantArrayType(ArrayTy);
```

Comment:
Does it make sense to assert if `CAT` is not null?

Reply:
I didn't think it was necessary since `visitArray `checked this.